docs/whitepaper.tex

```diff
@@ -157,80 +157,81 @@ \section{Implementation details}
-itself) and a datum with the liquidity token
-minting policy hash, and the pool validator script address. \\
+An alternative solution we propose this problem is the Config UTXO.
+The Config UTXO is a dedicated UTxO which is identified by a NFT, its datum 
+keeps track of all the script hashes involved in the protocol. Each script
+is then able refer to the script hashes in the Config UTXO in order to verify the other
+scripts. \\
 
-In order to make swaps and liquidity interactions possible without
-the config utxo some dependencies are passed as parameters as shown in Figure 1. \\
+A drawback of the Config UTXO is the greater transaction size and thus higher transaction fees.
+Therefore we make use of an hybrid approach between the naive and Config UTXO.
+In our case, the Config UTXO will contain the following data in the datum:
+the liquidity token minting policy hash and the pool address validator script hash.
+The other hashes will be passed as parameters. See Figure \ref{hashGraph}  \\
 
 \begin{figure}
-\scalebox{0.5}{\includegraphics{diagram}}
-\caption{
-	In the diagram above blue arrow represent passing hashes by
-	config utxo and red arrows represent passing hashes by script parameter.
-}
+  \scalebox{0.5}{\includegraphics{diagram}}
+  \caption{
+	  In the diagram above blue arrows represent passing hashes by
+	  Config UTXO and red arrows represent passing hashes by script parameter.
+  }
+  \label{hashGraph}
 \end{figure}
 
 It is important that the parameter sub-graph is acyclic so that
 the hashes are resolvable. Likewise it's important that
-the config utxo sub-graph only includes arrow from the pool id minting policy
-so that all other interactions can be performed without the config utxo.
+the Config UTXO sub-graph only includes arrows from the pool id minting policy
+so that all other interactions can be performed without the Config UTXO.
 
 
-\subsubsection{Config utxo addresses validator}
+\subsubsection{Config UTXO: Addresses Validator}
 
-The config utxo needs an address where it can be used as a reference
+The Config UTXO needs an address where it can be used as a reference
 input but can't be changed. Since use as a reference input doesn't
 invoke a validator this validator can just always fail.
 
-\subsubsection{Config utxo identifying NFT minting policy}
+\subsubsection{Config UTXO: Identifier NFT Minting Policy}
 
-The config utxo also needs an NFT to distinguish it from any
-invalid config utxos. This can just be a boiler plate NFT contract,
-so we can use the one already in the CAT (Cardano App Template).
+We need a way to distinguish and validate the Config UTXO from any
+other Config UTXOs. This can be implemented using a boilerplate NFT contract.
 
-\subsection{Pool ID minting policy}
+\subsection{DEX Pool: Identifier NFT Minting Policy}
 
-Each pool will be identified by an NFT. This allows anyone to verify that a UTXO
+Each pool will be identified by a NFT. This allows anyone to verify that a UTXO
```

Comment:
I think this is spelled "an NFT", actually.

Reply:
No, [see] (https://owl.purdue.edu/owl/general_writing/grammar/articles_a_versus_an.html)